lib/parsers/get-package-lock-dep-graph.ts

```diff
@@ -0,0 +1,167 @@
+  strict = true,
+): Promise<DepGraph> {
+  if (lockfile.type !== LockfileType.npm) {
+    throw new InvalidUserInputError(
+      'Unsupported lockfile provided. Please ' + 'provide `package-lock.json`.',
+    );
+  }
+  const packageLock = lockfile as PackageLock;
+
+  recursiveLinkRequire(packageLock.dependencies || {}, [packageLock]);
+
+  const rootPkgInfo = {
+    name: manifestFile.name,
+    version: manifestFile.version,
+  };
+
+  const pkgManager = {
+    name: LockfileType.npm,
+  };
+
+  const depGraphBuilder = new DepGraphBuilder(pkgManager, rootPkgInfo);
+
+  // const nodeVersion = manifestFile?.engines?.node;
+  // if (nodeVersion) {
+  //   _set(depTree, 'meta.nodeVersion', nodeVersion);
+  // }
+
+  // asked to process empty deps
+  if (_isEmpty(manifestFile.dependencies) && !includeDev) {
+    return depGraphBuilder.build();
+  }
+
+  const topLevelDeps: Dep[] = getTopLevelDeps(manifestFile, includeDev);
+
+  const devMemoizationMap: Map<PackageLockDep, NodeId> = new Map();
+  const prodMemoizationMap: Map<PackageLockDep, NodeId> = new Map();
+
+  for (const dep of topLevelDeps) {
+    const dependency = packageLock.dependencies?.[dep.name];
+    if (/^file:/.test(dep.version)) {
+      addTopLevelDep(dep);
+      continue;
+    }
+
+    const version = dependency?.version;
+    if (strict && !version) {
+      throw new OutOfSyncError(dep.name, LockfileType.npm);
+    }
+    const nodeId = addTopLevelDep(dep, version);
+    const isDev = !!dep.dev;
+
+    await recursiveRun(
+      depGraphBuilder,
+      dependency?.linkedRequires,
+      isDev,
+      isDev ? devMemoizationMap : prodMemoizationMap,
+      nodeId,
+    );
+  }
+
+  return depGraphBuilder.build();
+
+  function addTopLevelDep(dep: Dep, version: string = dep.version) {
+    const nodeId = getNodeId(dep.name, version);
+    depGraphBuilder.addPkgNode({ name: dep.name, version }, nodeId, {
+      labels: { scope: dep.dev ? Scope.dev : Scope.prod },
+    });
+    depGraphBuilder.connectDep(depGraphBuilder.rootNodeId, nodeId);
+    return nodeId;
+  }
+}
+
+function getNodeId(depName: string, version: string = '') {
+  return `${depName}@${version}|${Math.random()}`;
+}
+
+function recursiveLinkRequire(
+  lockfileDeps: PackageLockDeps,
+  ancestors: PackageLockDep[] = [],
+): void {
+  for (const dep of Object.values(lockfileDeps)) {
+    const currentAncestors = [dep].concat(ancestors);
+    if (dep.requires) {
+      dep.linkedRequires = {};
+      for (const key of Object.keys(dep.requires)) {
+        dep.linkedRequires[key] = find(key, currentAncestors);
+      }
+      delete dep.requires;
+    }
+
+    if (dep.dependencies) {
+      recursiveLinkRequire(dep.dependencies, currentAncestors);
+    }
+  }
+}
+
+function find(key: string, ancestors: PackageLockDep[] = []): PackageLockDep {
+  for (const dep of ancestors) {
+    if (dep.dependencies?.[key]) {
+      return dep.dependencies[key];
+    }
+  }
+  throw new Error('');
+}
+
+type NodeId = string;
```

Comment:
Should this be exported or moved to `types`?

Reply:
No, it's just an alias for internal types to use